sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -524,6 +535,9 @@ def mean_squared_scaled_error(
             "Another look at measures of forecast accuracy", International
             Journal of Forecasting, Volume 22, Issue 4.
     """
+    y_train = kwargs.pop("y_train", None)
+    if y_train is None:
+        raise ValueError("y_train must be supplied to compute this metric.")
```

Comment:
Since we're using this in multiple places, would it be worth it to write a helper function for this, e.g. `_get_y_train(**kwargs)` or even `_get_kwarg(**kwargs, "y_train")` so that we can use it for `y_pred_benchmark` too?

Reply:
Yep, I've moved all three of those line into a non-public function.